internal/kafka/internal/config/dataplane_cluster_config.go

```diff
@@ -240,23 +240,27 @@ func (conf *ClusterConfig) GetCapacityForRegionAndInstanceType(region, instanceT
 	return capacity
 }
 
-func (conf *ClusterConfig) IsNumberOfKafkaWithinClusterLimit(clusterId string, count int) bool {
-	if _, exist := conf.clusterConfigMap[clusterId]; exist {
-		limit := conf.clusterConfigMap[clusterId].KafkaInstanceLimit
+func (conf *ClusterConfig) IsNumberOfStreamingUnitsWithinClusterLimit(clusterID string, count int) bool {
+	if _, exist := conf.clusterConfigMap[clusterID]; exist {
+		limit := conf.clusterConfigMap[clusterID].KafkaInstanceLimit
 		return limit == -1 || count <= limit
 	}
+
+	// TODO - we've to consider returning false here if the cluster is not in the manual list.
```

Comment:
@pb82 I feel like this is the real issue. I feel like we should return `false` here since for all the clusters that are not in the manual list.